src/builder.rs

```diff
@@ -7,14 +7,45 @@ use super::process_impl;
 use super::native_mux_impl;
 
 use std::borrow::Cow;
+use std::env;
 use std::fs;
 use std::path::{Path, PathBuf};
 use std::process::Stdio;
 use std::str;
 
+use dirs::home_dir;
+use once_cell::sync::OnceCell;
 use tempfile::{Builder, TempDir};
 use tokio::process;
 
+/// The returned `&'static Path` can be coreced to any lifetime.
+fn get_default_control_dir<'a>() -> Result<&'a Path, Error> {
+    static DEFAULT_CONTROL_DIR: OnceCell<Option<Box<Path>>> = OnceCell::new();
+
+    DEFAULT_CONTROL_DIR
+        .get_or_try_init(|| {
+            if let Some(xdg_state_home) = env::var_os("XDG_STATE_HOME") {
```

Comment:
Why use this instead of [`dirs::state_dir`](https://docs.rs/dirs/latest/dirs/fn.state_dir.html)?

Reply:
Well, I used that so that `fs::create_dir_all` is only called for `$HOME/.local/state`.

I don't know whether it is a good idea to `fs::create_all($XDG_STATE_HOME)`.

Reply:
I don't think that should be a problem. The current process already runs as the user, so should be fine.

Reply:
OK, I have updated the `get_default_control_dir` to use `dirs::state_dir`.